### ISSUE TITLE
Ablation: ViT-S 20K with vs without scale embedding on LIDC

### ISSUE THREAD

**Issue by timlawrenz** [timlawrenz/DINO-X#4]
## Summary
Run a controlled ablation experiment to validate that the scale embedding helps (or at minimum doesn't hurt) on LIDC-IDRI.

## Experiment Design
| Arm | Scale Embedding | Spacing Data | Steps | Expected |
|-----|----------------|-------------|-------|----------|
| baseline | None | N/A | 20K | ratio ~311 (known) |
| scale_default | Yes | (1,1,1) for all | 20K | ratio ≈ 311 (no regression) |
| scale_real | Yes | Real DICOM spacing | 20K | ratio ≥ 311 (benefit) |

## Evaluation
- View retrieval ratio (primary metric)
- Malignancy linear probe AUC (secondary)
- t-SNE of scale embeddings for different spacings (visualization)

## Fleet Spec
Create `specs/zoo/scale_embedding_ablation.yaml` using the proven recipe (ViT-S, lr=2e-4, batch=64, cm=0.999, DINO+Gram+KoLeo).

## Depends On
- #2 (ViT integration)
- #3 (DICOM spacing extraction)

**Comment by timlawrenz** [comment]
## Technical Notes: Spacing Normalization & MLP Capacity

Two critical concerns for the ablation to produce valid results:

### 1. Spacing Normalization

The current `ScaleEmbedding` receives **raw mm values** directly: `(pixel_spacing_x, pixel_spacing_y, slice_thickness)`. In our MVP datasets:
- `pixel_spacing`: 0.46–0.98mm (LIDC-IDRI), 0.61–0.98mm (Pancreas-CT)
- `slice_thickness`: 0.625–5.0mm (LIDC-IDRI), 0.5–5.0mm (Pancreas-CT)

Slice thickness varies 10× while pixel spacing varies 2×. This means the first `nn.Linear(3, hidden)` layer receives inputs where one dimension dominates the gradient signal. For the ablation to be fair, we should **normalize spacing to zero mean / unit variance** using dataset statistics before the MLP:

```python
def forward(self, spacing: torch.Tensor) -> torch.Tensor:
    # Normalize using corpus statistics (computed from combined manifest)
    normalized = (spacing - self.spacing_mean) / self.spacing_std
    return self.mlp(normalized).unsqueeze(1)
```

Where `spacing_mean` and `spacing_std` are registered buffers computed from the training corpus's `SpacingStats`. This is analogous to how positional frequencies are fixed — the normalization constants are part of the model, not learned.

### 2. MLP Capacity

Current MLP: `Linear(3 → dim//4) → GELU → Linear(dim//4 → dim) → LayerNorm`

For ViT-Small (dim=384): hidden=96, giving 3×96 + 96×384 = 37K params. This is appropriately small (~0.15% of the 24.9M backbone). The zero-initialization of the output layer ensures it starts as a no-op and learns gradually.

The LayerNorm after the output projection further constrains the embedding magnitude. This design looks correct — the risk of dominating `patch_embed` signals is low.

### Ablation success criteria

The MVP fleet run (in progress now) is a 5K-step preview. For the full #4 ablation at 20K steps:
- `scale_default` (spacing=1,1,1): Must match baseline ratio within 5% (no regression test)
- `scale_real` (real DICOM spacing): Must meet or exceed baseline
- If `scale_real` < baseline at 20K steps → investigate spacing normalization first before concluding the embedding is harmful

**Comment by timlawrenz** [comment]
✅ Ablation completed (modified scope: 5K steps on two-organ combined data instead of 20K on LIDC-only).

**Results (2026-04-21):**
| Arm | Final Loss | View Retrieval Ratio |
|-----|-----------|---------------------|
| Baseline (no scale embed) | 8.992 | 5.0× random |
| Scale-aware (ScaleEmbedding) | **0.134** | 4.0× random |

The ScaleEmbedding breaks through the DINO entropy wall (ln(8192)=9.01) decisively. View retrieval ratios are comparable because both augmented views share identical spacing — need spacing-stratified eval for proper measurement.

Full results: `docs/EXPERIMENTS.md`, artifacts: `runs/mvp-two-organ/`
Commit: f7b4900
